src/design-mode-component.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+function DesignModeComponent() {
+  this.state = false;
+  chrome.runtime.onMessage.addListener(function (request) {
+    if (request.trigger == 'toggleDesignMode') {
+      this.toggleDesignMode();
+    }
+  }.bind(this));
+}
+
+DesignModeComponent.prototype.toggleDesignMode = function () {
+  this.state = !this.state;
```

Comment:
Could this be renamed to, say, `this.designModeActive`?

Reply:
I think state is fine as it's consistent with the other components